### DISCUSSION TITLE
[Feature Request] - The API https://api-e.ecoflow.com/ now supports Smart Plug, PowerStream Micro-inverter, Wave Air Conditioner and Glacier.

### DISCUSSION THREAD

**Discussion by Nid01** [tolwi/hassio-ecoflow-cloud#175] [Ideas]
Today the Ecofow Developer Group on Facebook announced the API https://api-e.ecoflow.com/ got updated to support the devices Smart Plug, PowerStream Micro-inverter, Wave Air Conditioner and Glacier.
Would it make sense to switch communications for the devices to this API so for example the PowerStream can be controlled without the need of ecoflow-powerstream-nodejs?

https://developer-eu.ecoflow.com/us/document/introduction
![image](https://github.com/tolwi/hassio-ecoflow-cloud/assets/30300755/501d9756-701b-4756-819a-719bdf925118)

I'll take my chances on testing how to use the API, but I'm not invested into software development too much so someone with better skills and interest might get it done sooner.

@tolwi Maybe for yourself it might be worth the effort switching the devices to the other API?

**Comment by Fyr42** [comment]
WOW at least they "opened" their API.

**Comment by Nid01** [comment]
Thanks to @svenerbe and @sani007zh for making this possible!
https://github.com/tolwi/hassio-ecoflow-cloud/issues/124#issuecomment-1944125107
https://github.com/tolwi/hassio-ecoflow-cloud/issues/124#issuecomment-1975284331
